src/VisualStudio/Core/Impl/CodeModel/FileCodeModel.cs

```diff
@@ -177,7 +177,13 @@ internal DocumentId GetDocumentId()
 
         internal void UpdateCodeElementNodeKey(AbstractKeyedCodeElement keyedElement, SyntaxNodeKey oldNodeKey, SyntaxNodeKey newNodeKey)
         {
-            var codeElement = _codeElementTable.Remove(oldNodeKey);
+            EnvDTE.CodeElement codeElement;
+            if (!_codeElementTable.TryGetValue(oldNodeKey, out codeElement))
+            {
+                throw new InvalidOperationException($"Could not find {oldNodeKey} in Code Model element table.");
+            }
+
+            _codeElementTable.Remove(oldNodeKey);
```

Comment:
The old code didn't check for null. It simply called ```ComAggregate.GetManagedObject(...)``` with the null value it received from ```_codeElement.Remove(...)```, which threw. The change is a bit more significant because I decided to call ```TryGetValue``` and bail out a bit early (with a good message) if ```oldNodeKey``` isn't in the table or doesn't have a code element.